ledger/error/src/main/scala/com/daml/error/BaseError.scala

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.error
+
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+
+/** The main error interface for everything that should be logged and notified.
+  *
+  * There are two ways to communicate an error to the user: write it into a log or send it as a string.
+  * In most cases, we'll do both: log the error appropriately locally and communicate it to the user
+  * by failing the API call with an error string.
+  */
+trait BaseError extends LocationMixin {
+
+  /** The error code, usually passed in as implicit where the error class is defined */
+  def code: ErrorCode
+
+  /** A human readable string indicating the error */
+  def cause: String
+
+  /** An optional argument to log exceptions
+    *
+    * If you want to log an exception as part of your error, then use the following example:
+    *
+    * object MyCode extends ErrorCode(id="SUPER_DUPER_ERROR") {
+    *   case class MyError(someString: String, throwable: Throwable) extends SomeInternalError(
+    *     cause = "Something failed with an exception bla",
+    *     throwableO = Some(throwable)
+    *   )
+    * }
+    */
+  def throwableO: Option[Throwable] = None
```

Comment:
We don't generally use this naming convention in the Daml SDK (`O` suffix for `Option`s).

Can we have simply `throwable` here, please? We already have e.g. `retryable` defined in this trait which returns an `Option[ErrorCategoryRetry]`.